core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -522,18 +587,18 @@ class ReplicaManagerTest {
     result
   }
 
-  private def setupReplicaManagerWithMockedPurgatories(timer: MockTimer): ReplicaManager = {
-    val props = TestUtils.createBrokerConfig(1, TestUtils.MockZkConnect)
+  private def setupReplicaManagerWithMockedPurgatories(timer: MockTimer, aliveBrokerIds: Seq[Int] = Seq(0, 1)): ReplicaManager = {
+    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect)
```

Comment:
Shouldn't we set the broker.id property since ReplicaManager needs config.brokerId?

Reply:
Yes, `createBrokerConfig` does it:

```scala
if (nodeId >= 0) props.put(KafkaConfig.BrokerIdProp, nodeId.toString)
```